### PR TITLE
Addded f/temel as template engine.

### PR DESCRIPTION
**Pull request by f**
Temel is a native template engine which doesn't have any syntax. Based on macros and very light-weight. It also supports custom tags.

### CODE REVIEW

**Comment by f** [comment]
Also formatted codebase using `crystal tool format` (Intended to make another PR but these were simple line deletions / additions, may be included with this one)

**Comment by makenowjust** [comment]
@f You should respect @jeromegn's coding style, in common-sense.

And `.cr` is too generally, I think. this extension will conflict probably.

**Comment by f** [comment]
@MakeNowJust I think you're right. I reverted.

**Comment by f** [comment]
@MakeNowJust These files are exactly crystal files. There aren't any parser/lexer in Temel, that's why I use `.cr` extension.

**Comment by makenowjust** [comment]
@f You are right, however the man, that makes new library using Crystal syntax as template, are also right. So, I think that is decided by user.

**Comment by f** [comment]
@MakeNowJust Hm, it makes sense. You win :) I'm going to update the PR now.

**Comment by makenowjust** [comment]
@f I think, `src/kilt/temal.cr` and some files are yours and maintained by you, so you can format these files. @jeromegn, how do you think?

And please see #3 and move `src/kilt/temal.cr` to `src/temal.cr` if you think `require "kilt/kilt/temal"` is wrong.

**Comment by f** [comment]
@MakeNowJust To make the codebase's code styling unique it should follow @jeromegn's code styling, I think.

Yeah, `"kilt/kilt/temel"` seems ugly. I am going to update the path.

**Comment by f** [comment]
@MakeNowJust But Travis will fail before your PR is merged. Maybe I should wait for #3 to be merged and update. WDYT?

**Comment by makenowjust** [comment]
@f No problem. I merged your pull request and #3 on local, then tested on this temporary branch, it was successed.

File: src/kilt/temel.cr [hunk 1]
```diff
@@ -0,0 +1,9 @@
+require "../kilt"
+require "temel"
+
+macro embed_temel(filename, __kilt_io__)
+  __kilt_io__ << {{ run("./helpers/temel_embedder.cr", filename) }}
```
Comment (jeromegn) [inline src/kilt/temel.cr]: I think the path to the embedder needs to be as if it was being included in an external project like: "kilt/kilt/helpers/temel_embedder" or else it won't work when people require it. I had to do that in slang.

Also, this new path will fail the tests :( unless you add symlinks to make that path accessible.

Might be best to add it to temel directly.

**Comment by jeromegn** [comment]
Thanks for that PR! I left a comment, but this is nice.

**Comment by jeromegn** [comment]
@f I merged PR #3 and so this will be fine with the new file locations. (sorry for all the changes with this PR opened)

**Comment by f** [comment]
@jeromegn I updated the PR. I tested with another project, `./helpers` thingy doesn't make any problem. `{{ run ... }}` macro runs the file from the path related to where it's run.

**Comment by f** [comment]
BTW. Travis is crying because of something else.

**Comment by jeromegn** [comment]
Hmm, interesting. I would've taught it would fail, I faced a while ago and filed an issue here: https://github.com/manastech/crystal/issues/2127

I'll have to trust you on this :)

**Comment by jeromegn** [comment]
Released as part of v0.3.2

**Pull request merged**